### PR TITLE
TritonPart V2 codes (cleaned codes and timing-driven flow)  && Fixed the off grid issue of Hier-RTLMP

### PR DESCRIPTION
**Pull request by ZhiangWang033**
Hi Matt,

I have cleaned the codes of TritonPart and finish the timing driven feature.
Also, the off grid issue of Hier-RTLMP has been fixed.

### CODE REVIEW

File: src/mpl2/src/hier_rtlmp.cpp [hunk 1]
```diff
@@ -317,6 +317,7 @@ void HierRTLMP::hierRTLMacroPlacer()
   //
   block_ = db_->getChip()->getBlock();
   dbu_ = db_->getTech()->getDbUnitsPerMicron();
+  manufacturing_unit_ = db_->getTech()->getManufacturingGrid();
```
Comment (maliberty) [inline src/mpl2/src/hier_rtlmp.cpp:320]: Though most LEF has this value it is possible for it to be unset.  You should check hasManufacturingGrid().  If none is present you could use 1dbu

File: src/par/CMakeLists.txt [hunk 2]
```diff
@@ -35,6 +35,18 @@
 
 include("openroad")
 
+option(LOAD_CPLEX "Load CPLEX" ON)
```
Comment (maliberty) [inline src/par/CMakeLists.txt:38]: You should either make this conditional on finding cplex or off by default.

File: src/par/CMakeLists.txt [hunk 3]
```diff
@@ -35,6 +35,18 @@
...
+if (LOAD_CPLEX)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_MODULE_PATH               "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
```
Comment (maliberty) [inline src/par/CMakeLists.txt:41]: it would be better to use ```list(APPEND ...``` than to overwrite the value.

**Comment by ZhiangWang033** [comment]
Fixed.  Turn off CPLEX.

File: src/par/CMakeLists.txt [hunk 4]
```diff
@@ -35,6 +35,18 @@
...
+  set(CMAKE_CXX_FLAGS                 "${CMAKE_CXX_FLAGS} -DIL_STD -m64 -Wall -fPIC")
+  set(CMAKE_INCLUDE_SYSTEM_FLAG_CXX   "-isystem ")
+  set(LINKER_OPTIONS                  "-Wl,--export-dynamic")
```
Comment (maliberty) [inline src/par/CMakeLists.txt:44]: Don't set these as they are global.  Use target_compile_options and target_link_options which are specific to a particular target.  

-m64 & -Wall are not necessary to set per project.

File: src/par/CMakeLists.txt [hunk 5]
```diff
@@ -35,6 +35,18 @@
...
+  find_package(Cplex)
+  include_directories(SYSTEM ${CPLEX_INCLUDE_DIRS})
```
Comment (maliberty) [inline src/par/CMakeLists.txt:46]: Use target_include_directories(... SYSTEM ...

**Comment by ZhiangWang033** [comment]
Hi Matt, I am not familiar with CMAKE file.

Could you please help me modify this file ?   I have give you access for
this repository,

Thanks,
Zhiang

**Comment by maliberty** [comment]
Ok I'll fix it up.

Matt

**Comment by maliberty** [comment]
Replaced by https://github.com/The-OpenROAD-Project-private/OpenROAD/pull/1718

**Pull request closed without merging**